### PR TITLE
T19: Correctly place Snackbar depending on currently displayed fragments

### PR DESCRIPTION
**Pull request by KevinBoulongne**
Depends on #1595

### CODE REVIEW

File: app/src/main/java/com/infomaniak/mail/ui/MainActivity.kt [hunk 1]
```diff
@@ -377,8 +382,26 @@ class MainActivity : BaseActivity() {
...
         fun getAnchor(): View? = when (navController.currentDestination?.id) {
-            R.id.threadListFragment -> findViewById(R.id.newMessageFab)
+            R.id.threadListFragment -> {
+                val threadListFragment = this@MainActivity.currentFragment as? ThreadListFragment
+                if (threadListFragment?.isOnlyRightShown() == true) {
+                    val threadFragment = threadListFragment.binding.threadHostFragment.getFragment<ThreadFragment?>()
+                    threadFragment?.binding?.quickActionBar
+                } else {
+                    threadListFragment?.binding?.newMessageFab
+                }
```
[the remark this answers is not in the document]
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/MainActivity.kt]: Done !

File: app/src/main/java/com/infomaniak/mail/ui/MainActivity.kt [hunk 2]
```diff
@@ -377,8 +382,26 @@ class MainActivity : BaseActivity() {
...
+
         fun getAnchor(): View? = when (navController.currentDestination?.id) {
-            R.id.threadListFragment -> findViewById(R.id.newMessageFab)
+            R.id.threadListFragment -> {
+                val threadListFragment = this@MainActivity.currentFragment as? ThreadListFragment
+                if (threadListFragment?.isOnlyRightShown() == true) {
+                    val threadFragment = threadListFragment.binding.threadHostFragment.getFragment<ThreadFragment?>()
+                    threadFragment?.binding?.quickActionBar
+                } else {
+                    threadListFragment?.binding?.newMessageFab
```
[the remark this answers is not in the document]
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/MainActivity.kt]: Done !

File: app/src/main/java/com/infomaniak/mail/ui/MainActivity.kt [hunk 3]
```diff
@@ -377,8 +381,10 @@ class MainActivity : BaseActivity() {
     }
 
     private fun setupSnackBar() {
+
         fun getAnchor(): View? = when (navController.currentDestination?.id) {
-            R.id.threadListFragment -> findViewById(R.id.newMessageFab)
+            R.id.threadListFragment -> (currentFragment as? ThreadListFragment)?.getAnchor()
+            R.id.searchFragment -> (currentFragment as? SearchFragment)?.getAnchor()
```
Comment (FabianDevel) [inline app/src/main/java/com/infomaniak/mail/ui/MainActivity.kt:387]: as threadListFragment and searchFragment are both TwoPaneFragment, we could regroup this with an abstract getAnchor() in TwoPaneFragment so we only have 1 case

**Review by FabianDevel — CHANGES_REQUESTED** [review]

**Review by LunarX — APPROVED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/MainActivity.kt [hunk 4]
```diff
@@ -342,13 +348,9 @@ class MainActivity : BaseActivity() {
...
-                ?.primaryNavigationFragment
-
-            if (currentFragment is TwoPaneFragment) {
-                currentFragment.handleOnBackPressed()
+            val fragment = currentFragment
+            if (fragment is TwoPaneFragment) {
+                fragment.handleOnBackPressed()
             } else {
                 navController.popBackStack()
             }
```
Comment (FabianDevel) [inline app/src/main/java/com/infomaniak/mail/ui/MainActivity.kt:356]: Oneline this if

**Review by FabianDevel — CHANGES_REQUESTED** [review]

**Review by FabianDevel — APPROVED** [review]

**Pull request merged**